ibm/resource_ibm_cis_firewall.go

```diff
@@ -88,220 +257,570 @@ func resourceIBMCISFirewallRecord() *schema.Resource {
-			}
-			var urls = make([]string, 0)
-			if u, ok := lockdownMap["urls"]; ok && u != nil {
-				for _, url := range u.([]interface{}) {
-					urls = append(urls, fmt.Sprintf("%v", url))
-				}
-				newRecord.Urls = urls
-			}
-			var configurationList = make([]v1.Configuration, 0)
-			if res, ok := lockdownMap["configurations"]; ok {
-				configurations := res.([]interface{})
-				for _, c := range configurations {
-					r, _ := c.(map[string]interface{})
-					if target, value := r["target"], r["value"]; target != nil && value != nil {
-						configurationRecord := v1.Configuration{}
-						configurationRecord.Target, configurationRecord.Value = target.(string), value.(string)
-						configurationList = append(configurationList, configurationRecord)
-					}
-				}
-				newRecord.Configurations = configurationList
-			}
+		opt := cisClient.NewCreateZoneLockdownRuleOptions()
+		// not able to check bool variable availability
+		v, _ := lockdown[cisFirewallLockdownPaused]
+		opt.SetPaused(v.(bool))
+		if v, ok := lockdown[cisFirewallLockdownDesc]; ok && v.(string) != "" {
+			opt.SetDescription(v.(string))
 		}
-	}
+		if v, ok := lockdown[cisFirewallLockdownPriority]; ok && v.(int) > 0 {
+			opt.SetPriority(int64(v.(int)))
+		}
+		urls := expandStringList(lockdown[cisFirewallLockdownURLs].([]interface{}))
+		configurations, err := expandLockdownsTypeConfiguration(
+			lockdown[cisFirewallLockdownConfigurations].([]interface{}))
+		if err != nil {
+			return err
+		}
+		opt.SetUrls(urls)
+		opt.SetConfigurations(configurations)
 
-	var recordPtr *v1.FirewallRecord
-	recordPtr, err = cisClient.Firewall().CreateFirewall(cisID, zoneID, firewallType, newRecord)
-	if err != nil {
-		return fmt.Errorf("Failed to create Firewall: %s", err)
-	}
+		cisClient.Crn = core.StringPtr(crn)
+		cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+		result, response, err := cisClient.CreateZoneLockdownRule(opt)
+		if err != nil {
+			log.Printf("Create zone firewall lockdown failed: %v", response)
+			return err
+		}
+		log.Printf("Zone Firewall Lockdown created successfully : %s", *result.Result.ID)
+		d.SetId(convertCisToTfFourVar(firewallType, *result.Result.ID, zoneID, crn))
 
-	// In the Event that the API returns an empty Firewall, we verify that the
-	// ID returned is not the default ""
+	} else if firewallType == cisFirewallTypeAccessRules {
 
-	record := *recordPtr
+		cisClient, err := meta.(ClientSession).CisAccessRuleClientSession()
+		if err != nil {
+			return err
+		}
+		cisClient.Crn = core.StringPtr(crn)
+		cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+		accessRule := d.Get(cisFirewallAccessRule).([]interface{})[0].(map[string]interface{})
 
-	if record.ID == "" {
-		return fmt.Errorf("Failed to find record in Create response; Record was empty")
-	}
+		mode := accessRule[cisFirewallAccessRuleMode].(string)
+
+		configList := accessRule[cisFirewallAccessRuleConfiguration].([]interface{})
+		if len(configList) > 1 {
```

Comment:
Why is this check provided here?
I believe MaxItem:1 in schema does this work

Reply:
removed